scripts/js_test.py

```diff
@@ -0,0 +1,492 @@
+        else:
+            msg = colorize('green', "\n\n{bar}\nNo test failures ".format(bar="=" * 48))
+
+        print(msg)
+
+    def run(self):
+        """
+        Runs the tests in the suite while tracking and reporting failures.
+        """
+        self.run_suite_tests()
+
+        # if tasks.environment.dry_run:
+        #     return
+
+        self.report_test_results()
+
+        if self.failed_suites:
+            sys.exit(1)
+
+
+class JsTestSuite(TestSuite):
+    """
+    A class for running JavaScript tests.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.run_under_coverage = kwargs.get('with_coverage', True)
+        self.mode = kwargs.get('mode', 'run')
+        self.report_dir = Env.JS_REPORT_DIR
+        self.opts = kwargs
+
+        suite = args[0]
+        self.subsuites = self._default_subsuites if suite == 'all' else [JsTestSubSuite(*args, **kwargs)]
+
+    def __enter__(self):
+        super().__enter__()
+        self.report_dir.makedirs_p()
+        # self.report_dir.mkdir(exist_ok=True)
+        # if not self.skip_clean:
+        # test_utils.clean_test_files()
+
+        # if self.mode == 'run' and not self.run_under_coverage:
+        # test_utils.clean_dir(self.report_dir)
+
+    @property
+    def _default_subsuites(self):
+        """
+        Returns all JS test suites
+        """
+        return [JsTestSubSuite(test_id, **self.opts) for test_id in Env.JS_TEST_ID_KEYS if test_id != 'jest-snapshot']
+
+
+class JsTestSubSuite(TestSuite):
+    """
+    Class for JS suites like cms, cms-squire, lms, common,
+    common-requirejs and xmodule
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.test_id = args[0]
+        self.run_under_coverage = kwargs.get('with_coverage', True)
+        self.mode = kwargs.get('mode', 'run')
+        self.port = kwargs.get('port')
+        self.root = self.root + ' javascript'
+        self.report_dir = Env.JS_REPORT_DIR
+
+        try:
+            self.test_conf_file = Env.KARMA_CONFIG_FILES[Env.JS_TEST_ID_KEYS.index(self.test_id)]
+        except ValueError:
+            self.test_conf_file = Env.KARMA_CONFIG_FILES[0]
+
+        self.coverage_report = self.report_dir / f'coverage-{self.test_id}.xml'
+        self.xunit_report = self.report_dir / f'javascript_xunit-{self.test_id}.xml'
+
+    @property
+    def cmd(self):
+        """
+        Run the tests using karma runner.
+        """
+        cmd = [
+            "node",
+            "--max_old_space_size=4096",
+            "node_modules/.bin/karma",
+            "start",
+            self.test_conf_file,
+            "--single-run={}".format('false' if self.mode == 'dev' else 'true'),
+            "--capture-timeout=60000",
+            f"--junitreportpath={self.xunit_report}",
+            f"--browsers={Env.KARMA_BROWSER}",
+        ]
```

Comment:
@kdmccormick There is one thing if could guide me, When i run the command directly from `Make` file and in case it can not start Firefox browser the process does not kill and hangs on for an indefinite time. 

Command in Make file:
```
@node --max_old_space_size=4096 node_modules/.bin/karma start cms/static/karma_cms.conf.js \
		--single-run=false \
		--capture-timeout=60000 \
		--browsers=FirefoxNoUpdates
```

Console warning:

> 04 11 2024 14:18:37.622:INFO [launcher]: Starting browser Firefox
> 04 11 2024 14:19:37.633:WARN [launcher]: Firefox have not captured in 60000 ms, killing.
> 04 11 2024 14:19:39.680:WARN [launcher]: Firefox was not killed in 2000 ms, sending SIGKILL.
> 04 11 2024 14:19:41.688:WARN [launcher]: Firefox was not killed by SIGKILL in 2000 ms, continuing.

In the python code they are manually killing the process like
https://github.com/openedx/edx-platform/blob/d0dbb8d55df6a9cb3217a48e698d2b9fb92920ad/pavelib/utils/process.py#L16